### ISSUE TITLE
EntityView class not found when running Quarkus native image

### ISSUE THREAD

**Issue by kekbur** [Blazebit/blaze-persistence#1130]
### Description
The exception `javassist.NotFoundException: org.acme.PersonView` is thrown when using blaze-persistence integration with Quarkus and running the application as a native image built by GraalVM. The issue is not present when running on normal JVM.

### Expected behavior
The `NotFoundException` is not thrown, class [org.acme.PersonView](https://github.com/kekbur/blaze-persistence-quarkus-native-image-test/blob/master/src/main/java/org/acme/PersonView.java) is found.

### Actual behavior
The exception is thrown generating the following error message:

```
2020-07-11 14:34:35,175 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /persons failed, error id: 55f4c856-f312-498d-b318-ea7072e421c3-1: org.jboss.resteasy.spi.UnhandledException: java.lang.RuntimeException: Probably we did something wrong, please contact us if you see this message.
        at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:216)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:515)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:259)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:160)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:163)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:245)
        at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:132)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.access$000(VertxRequestHandler.java:37)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:94)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
        at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
        at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
Caused by: java.lang.RuntimeException: Probably we did something wrong, please contact us if you see this message.
        at com.blazebit.persistence.view.impl.proxy.ProxyFactory.createProxyClass(ProxyFactory.java:565)
        at com.blazebit.persistence.view.impl.proxy.ProxyFactory.getProxy(ProxyFactory.java:240)
        at com.blazebit.persistence.view.impl.proxy.ProxyFactory.getProxy(ProxyFactory.java:165)
        at com.blazebit.persistence.view.impl.proxy.TupleConstructorReflectionInstantiator.<init>(TupleConstructorReflectionInstantiator.java:40)
        at com.blazebit.persistence.view.impl.proxy.AbstractReflectionInstantiator.createInstantiator(AbstractReflectionInstantiator.java:93)
        at com.blazebit.persistence.view.impl.objectbuilder.ViewTypeObjectBuilderTemplate.createInstantiator(ViewTypeObjectBuilderTemplate.java:381)
        at com.blazebit.persistence.view.impl.objectbuilder.ViewTypeObjectBuilderTemplate.<init>(ViewTypeObjectBuilderTemplate.java:317)
        at com.blazebit.persistence.view.impl.objectbuilder.ViewTypeObjectBuilderTemplate.<init>(ViewTypeObjectBuilderTemplate.java:121)
        at com.blazebit.persistence.view.impl.objectbuilder.ViewTypeObjectBuilderTemplate$Key.createValue(ViewTypeObjectBuilderTemplate.java:1377)
        at com.blazebit.persistence.view.impl.EntityViewManagerImpl.getTemplate(EntityViewManagerImpl.java:1370)
        at com.blazebit.persistence.view.impl.EntityViewManagerImpl.createObjectBuilder(EntityViewManagerImpl.java:1354)
        at com.blazebit.persistence.view.impl.EntityViewManagerImpl.applyObjectBuilder(EntityViewManagerImpl.java:1320)
        at com.blazebit.persistence.view.impl.EntityViewSettingHelper.apply(EntityViewSettingHelper.java:104)
        at com.blazebit.persistence.view.impl.EntityViewManagerImpl.applySetting(EntityViewManagerImpl.java:1219)
        at com.blazebit.persistence.view.impl.EntityViewManagerImpl.find(EntityViewManagerImpl.java:773)
        at com.blazebit.persistence.view.impl.EntityViewManagerImpl.find(EntityViewManagerImpl.java:763)
        at com.blazebit.persistence.integration.quarkus.runtime.DefaultEntityViewManagerProducer_ProducerMethod_produceEntityViewManager_dde1367970cb94209df8af3dd492943b7e1a2f61_ClientProxy.find(DefaultEntityViewManagerProducer_ProducerMethod_produceEntityViewManager_dde1367970cb94209df8af3dd492943b7e1a2f61_ClientProxy.zig:622)
        at org.acme.PersonResource.createPerson(PersonResource.java:33)
        at org.acme.PersonResource_Subclass.createPerson$$superaccessor1(PersonResource_Subclass.zig:201)
        at org.acme.PersonResource_Subclass$$function$$1.apply(PersonResource_Subclass$$function$$1.zig:29)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:127)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:100)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.doIntercept(TransactionalInterceptorRequired.java:32)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.intercept(TransactionalInterceptorBase.java:53)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.intercept(TransactionalInterceptorRequired.java:26)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired_Bean.intercept(TransactionalInterceptorRequired_Bean.zig:339)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
        at org.acme.PersonResource_Subclass.createPerson(PersonResource_Subclass.zig:159)
        at java.lang.reflect.Method.invoke(Method.java:566)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:167)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
        at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:638)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:504)
        at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:454)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:456)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:417)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:391)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:68)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:488)
        ... 19 more
Caused by: javassist.NotFoundException: org.acme.PersonView
        at javassist.ClassPool.get(ClassPool.java:452)
        at com.blazebit.persistence.view.impl.proxy.ProxyFactory.createProxyClass(ProxyFactory.java:343)
        ... 61 more
```

#### Clone the test project

`git clone https://github.com/kekbur/blaze-persistence-quarkus-native-image-test.git && cd blaze-persistence-quarkus-native-image-test`

#### Build a native image with GraalVM and Docker

`docker build -f docker/Dockerfile.native -t blaze-persistence-quarkus-native-image:test .`

#### Start PostgreSQL

Unfortunately reproducing the issue requires that the application can connect to a PostgreSQL server.

`docker run --name postgres -p 5432:5432 -e POSTGRES_PASSWORD=password -d postgres`

#### Run the built image

Modify the datasource URL and link parameter to suit your environment if needed.

`docker run --link postgres -e "QUARKUS_DATASOURCE_URL=jdbc:postgresql://postgres:5432/postgres" --rm -p 8080:8080 blaze-persistence-quarkus-native-image:test`

#### Send an HTTP request to the /persons resource

`curl "http://localhost:8080/persons"`

### Environment

Version:            1.5.0-Alpha5
JPA-Provider:       Hibernate 5.4.18.Final
DBMS:               PostgreSQL
Application Framework: Quarkus

**Comment by beikov** [comment]
You seem to be missing the dependency for the annotation processor:

```xml
            <dependencies>
                <dependency>
                    <groupId>${project.groupId}</groupId>
                    <artifactId>blaze-persistence-entity-view-processor</artifactId>
                    <version>${project.version}</version>
                    <scope>provided</scope>
                </dependency>
            </dependencies>
```

**Comment by beikov** [comment]
I just checked, we didn't mention this in our or the Quarkus documentation. The use of the annotation processor is a requirement when using native compilation.

**Comment by beikov** [comment]
We will fix this with #1131 so that others don't run into this. Thanks for the report!
